### PR TITLE
Fixes #34395 - Make ansible-runner run without a PTY

### PR DESCRIPTION
**Pull request by adamruzicka**

### CODE REVIEW

File: lib/smart_proxy_ansible/runner/ansible_runner.rb [hunk 1]
```diff
@@ -34,10 +34,23 @@ def start
...
+        @command_pid = spawn(*command, :out => w, :err => w, :in => '/dev/null')
+        @command_out = r
+        w.close
+      rescue Errno::ENOENT => e
+        publish_exception("Error running command '#{command.join(' ')}'", e)
+      end
+
       def refresh
-        return unless super
+        super
```
Comment (ezr-ondrej) [inline lib/smart_proxy_ansible/runner/ansible_runner.rb:48]: Does super do something useful for us? Is it to close the Pipe when the command finishes?

Comment (adamruzicka) [reply lib/smart_proxy_ansible/runner/ansible_runner.rb:48]: It watches the process running as `@command_pid`, reads its output (which we later discard) and finally reaps the process

**Review by ezr-ondrej — APPROVED** [review]
I'm little bit lost how that pipe works with the refresh together, but it does and I do not see any processes leftovers, so I guess we are good.
It works smoothly for me. Let's get that in :+1:

**Comment by ezr-ondrej** [comment]
CP 3.3: c8289c2, 610e16f

**Pull request merged**